youngjoo00/week_05/BOJ_26090_완전한 수열.swift

```diff
@@ -0,0 +1,51 @@
+// BOJ_26090_완전한 수열.swift
+// 수열의 연속 부분 수열중, 완전한 수열의 갯수를 찾아서 수열의 안정도 출력
+// 완전한 수열의 조건
+// 1. 수열의 길이가 소수
+// 2. 수열의 합이 소수
+
+let N = Int(readLine()!)!
+let sequence = readLine()!.split(separator: " ").map { Int($0)! }
+
+// 총 수열의 갯수와 수열의 합 중 뭐가 더 큰지 비교하기 위해 합을 구함
+let sum = sequence.reduce(0, +)
+
+var primes: Set<Int> = []
+var result = 0
+
+// 에라토스테네스의 체를 사용하여 소수 구하기
+func getPrimes(_ limit: Int) -> Set<Int> {
+    var primes: Set<Int> = []
+    var number: [Bool] = [false, false] + repeatElement(true, count: limit-1)
+    
+    for i in 2...limit {
+        if number[i] {
+            primes.insert(i)
+            for j in stride(from: i, through: limit, by: i) {
+                number[j] = false
+            }
+        }
+    }
+    
+    return primes
+}
+
+// 소수 판별
+if 500 > sum {
+    primes = getPrimes(500)
+} else if sum >= 2 {
+    primes = getPrimes(sum)
+}
```

Comment:
위의 if 500 > sum 에 
else if sum >= 2 범위가 포함되지 않나요?
아래의 else if 문은 500보다 큰 경우를 계산하기 위함인건지 궁금합니다

Reply:
2...limit 까지 반복을 할 때 소수 판별을 효율적으로 하기위한 조건문입니다.

우선, 아래의 수열의 합을 구합니다.
// 총 수열의 갯수와 수열의 합 중 뭐가 더 큰지 비교하기 위해 합을 구함
let sum = sequence.reduce(0, +)

그런 다음, sum이 500보다 작은 경우, 500까지의 소수를 찾습니다.
이 경우, 수열의 합은 수열의 길이를 초과하지 않으며 최악의 경우를 생각했을 때,
최대 500까지의 소수를 찾으면 됩니다.

그러나 sum이 500보다 크거나 같은 경우, 수열의 합은 수열의 길이를 초과했다는 의미이며,
sum까지의 소수를 찾는 것입니다.

Reply:
```
그런 다음, sum이 500보다 작은 경우, 500까지의 소수를 찾습니다.
이 경우, 수열의 합은 수열의 길이를 초과하지 않으며 최악의 경우를 생각했을 때,
최대 500까지의 소수를 찾으면 됩니다.
```
이 부분에 반례가 있는게,
`2,3,4` 수열이 있다면, 길이는 3이고, 합은 9이니 수열의 합이 수열의 길이를 초과하는 경우가 있습니다.

수열의 길이와 수열의 합을 비교하는 아이디어가 시간 효율성 때문이라고 생각이 드는데, 
시간 효율성 때문이 맞다면,
다음과 같이 N과 sum을 비교하는 것이 좋지 않을까 싶습니다.

```suggestion
// 소수 판별
if N > sum {
    primes = getPrimes(N)
} else  {
    primes = getPrimes(sum)
}
```